From 052818e91314b797d96d87672c60f6784160f935 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Mon, 25 Mar 2024 10:50:59 -0700
Subject: [PATCH 2/2] Add missing STATX_ATTR_* defines from musl sys/stat.h

These defines are not provided by sys/stat.h on musl but are
in glibc's sys/stat.h, therefore to compile the tests on musl
based systems add the missing defines from linux/stat.h
if they are not provided

Upstream-Status: Submitted [https://lore.kernel.org/fstests/20240325175323.3450750-2-raj.khem@gmail.com/T/#u]
Signed-off-by: Khem Raj <raj.khem@gmail.com>
---
 src/statx.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/src/statx.h b/src/statx.h
index 3f239d79..12599f5b 100644
--- a/src/statx.h
+++ b/src/statx.h
@@ -12,6 +12,18 @@
 #define AT_STATX_DONT_SYNC      0x4000  /* - Don't sync attributes with the server */
 #endif
 
+#ifndef STATX_ATTR_COMPRESSED
+#define STATX_ATTR_COMPRESSED		0x00000004 /* [I] File is compressed by the fs */
+#define STATX_ATTR_IMMUTABLE		0x00000010 /* [I] File is marked immutable */
+#define STATX_ATTR_APPEND		0x00000020 /* [I] File is append-only */
+#define STATX_ATTR_NODUMP		0x00000040 /* [I] File is not to be dumped */
+#define STATX_ATTR_ENCRYPTED		0x00000800 /* [I] File requires key to decrypt in fs */
+#define STATX_ATTR_AUTOMOUNT		0x00001000 /* Dir: Automount trigger */
+#define STATX_ATTR_MOUNT_ROOT		0x00002000 /* Root of a mount */
+#define STATX_ATTR_VERITY		0x00100000 /* [I] Verity protected file */
+#define STATX_ATTR_DAX			0x00200000 /* File is currently in DAX state */
+#endif
+
 #ifndef AT_NO_AUTOMOUNT
 #define AT_NO_AUTOMOUNT		0x800	/* Suppress terminal automount traversal */
 #endif
-- 
2.44.0

